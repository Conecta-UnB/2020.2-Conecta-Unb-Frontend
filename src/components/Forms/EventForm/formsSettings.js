import * as Yup from 'yup';

export const initialValues = {
  titulo: '',
  organizador: '',
  date: '',
  descricao: '',
};

export const EventSchema = Yup.object().shape({
  titulo: Yup.string().required('É necessário informar um título'),
  organizador: Yup.string().required('É necessário indicar um organizador'),
  date: Yup.date().required('É necessário informar uma data'),
  descricao: Yup.string().required('É necessário indicar a descrição do evento'),
});

export const handleSubmit = async (values, setSubmitting, createEvent) => {
  setSubmitting(false);
  await createEvent({
    titulo: values.titulo,
    organizador: values.organizador,
    date: values.date,
    descricao: values.descricao,
  });
};
