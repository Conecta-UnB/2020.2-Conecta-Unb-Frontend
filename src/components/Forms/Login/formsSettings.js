import * as Yup from 'yup';

export const initialValues = {
  email: '',
  senha: '',
};

export const SignupSchema = Yup.object().shape({
  email: Yup
    .string()
    .email('Email inváilido')
    .required('É necessário indicar um email'),
  senha: Yup
    .string()
    .min(6, 'A senha deve ter pelo menos 6 dígitos')
    .required('É necessário criar uma senha'),

});

export const handleSubmit = async (values, setSubmitting, login) => {
  setSubmitting(false);
  await login({
    email: values.email,
    senha: values.senha,
  });
};
