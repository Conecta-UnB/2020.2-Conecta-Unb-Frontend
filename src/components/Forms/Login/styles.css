.containerLoginForm{
  background-color: #0D8447;
  width: 350px;
  height: 380px;
  border-radius: 20px;
  display: block;
}

.containerLoginForm img{
  display: block;
  margin-left: auto;
  margin-right: auto;
  width: 100px;
  height: 100px;
  border-radius: 50px;
}

.containerLoginForm div{
  align-self: center;
  padding-top: 5%;
}

.containerLoginForm input{
  width: 90%;
  margin-left: 5%;
  margin-right: 5%;
  height: 40px;
  border-radius: 20px;
}
.containerLoginForm label{
  font-size: 20px;
  margin-left: 5%;
  color: #ffff;
  font-weight: bold;
}

.containerLoginForm br{
  margin-top: 8%;
}

.containerLoginForm button{
  background: #0260EE;
  align-self: center;
  display: block;
  margin-top: 5%;
  margin-left: auto;
  margin-right: auto;
  width: 30%;
  height: 30px;
  border-radius: 8px;
  border: none;
  color: #ffff;
  font-weight: bold;
}