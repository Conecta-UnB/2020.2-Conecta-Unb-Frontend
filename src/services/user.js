export async function createUser(Host, user) {
  try {
    const response = await fetch(`${Host}/api/user/create`, {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        nome: user.nome,
        matricula: user.matricula,
        email: user.email,
        senha: user.senha,
        telefone: user.telefone,
      }),
    });
    const data = await response.json();
    console.log('Success: ', data);
    return data;
  } catch (error) {
    console.log('error ', error);
    return error;
  }
}
export async function getUser(Host, matricula) {
  const response = await fetch(`${Host}/api/user/${matricula}`);
  try {
    const data = await response.json;
    return data.data.user;
  } catch (error) {
    return error;
  }
}
