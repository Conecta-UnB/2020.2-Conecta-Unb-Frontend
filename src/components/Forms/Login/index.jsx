/* eslint-disable react/button-has-type */
/* eslint-disable jsx-a11y/label-has-associated-control */
import React from 'react';
import logo from '../../../assets/logo.jpeg';
import './styles.css';

export default function LoginForm() {
  return (
    <div className="containerLoginForm">
      <div>
        <img src={logo} alt="logoImage" />
        <form>
          <label>
            E-mail:
          </label>
          <input type="email" autoComplete="off" />
          <label>
            Senha:
          </label>
          <input type="password" autoComplete="off" />
          <button type="submit">
            Entrar
          </button>
        </form>
      </div>
    </div>
  );
}
