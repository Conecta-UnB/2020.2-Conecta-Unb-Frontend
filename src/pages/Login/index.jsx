import React from 'react';
import LoginForm from '../../components/Forms/Login';
import './styles.css';

export default function LoginPage() {
  return (
    <div className="underHeader">
      <LoginForm />
    </div>
  );
}
