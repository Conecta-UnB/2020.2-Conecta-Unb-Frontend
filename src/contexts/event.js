import React, { createContext, useState } from 'react';
import * as EventService from '../services/event';

export const EventContext = createContext({});

// eslint-disable-next-line react/prop-types
export default function EventProvider({ children }) {
  const [allEvents, setAllEvents] = useState('');
  const [events] = useState('');
  const Host = 'http://localhost:3333';

  async function createEvent(event) {
    console.log('ola', event);
    const response = await EventService.createEvent(Host, event);
    return response;
  }
  async function updateEvent(event) {
    console.log('ola', event);
    const response = await EventService.updateEvent(Host, event);
    return response;
  }

  async function deleteEvent(event) {
    console.log('ola', event);
    const response = await EventService.deleteEvent(Host, event);
    return response;
  }

  async function readEvent(event) {
    console.log('ola', event);
    const response = await EventService.readEvent(Host, event);
    return response;
  }

  async function readAllEvents() {
    const response = await EventService.readAllEvents(Host);
    setAllEvents(response.data);
    console.log(response.data);
    console.log('estou aqui');
    return response;
  }

  return (
    <EventContext.Provider
      value={{
        events,
        allEvents,
        createEvent,
        updateEvent,
        deleteEvent,
        readAllEvents,
        readEvent,
      }}
    >
      {children}
    </EventContext.Provider>
  );
}
