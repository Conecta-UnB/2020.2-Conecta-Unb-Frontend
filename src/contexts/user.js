import React, { createContext } from 'react';
import * as UserService from '../services/user';

export const UserContext = createContext({});

// eslint-disable-next-line react/prop-types
export default function UserProvider({ children }) {
  // const [user, setUser] = useState({});
  const Host = 'http://localhost:3333';
  async function createUser(user) {
    console.log('ola', user);
    const response = await UserService.createUser(Host, user);
    return response;
  }

  async function login(user) {
    const response = await UserService.login(Host, user);
    return response;
  }

  return (
    <UserContext.Provider
      value={{
        createUser,
        login,
      }}
    >
      {children}
    </UserContext.Provider>
  );
}
