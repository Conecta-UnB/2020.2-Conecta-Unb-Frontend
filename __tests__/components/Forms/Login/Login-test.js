import React from 'react';
import { render, waitFor, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { MemoryRouter } from 'react-router-dom';
import * as formSettings from '../../../../src/components/Forms/Login/formsSettings';
import { UserContext } from '../../../../src/contexts/user';
import LoginForm from '../../../../src/components/Forms/Login';

describe('testing loginforms', () => {
  test('must match snapshot', () => {
    const tree = render(
      <MemoryRouter initialEntries={[{ pathname: '/login' }]}>
        <LoginForm />
      </MemoryRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
  test('test if the form works', async () => {
    const login = jest.fn();
    formSettings.handleSubmit = jest.fn();
    render(
      <UserContext.Provider value={{ login }}>
        <MemoryRouter initialEntries={[{ pathname: '/login' }]}>
          <LoginForm />
        </MemoryRouter>
      </UserContext.Provider>,
    );
    userEvent.type(screen.getByLabelText(/e-mail/i, { selector: 'input' }), 'senhor_teste@gmail.com');
    userEvent.type(screen.getByLabelText(/senha/i, { selector: 'input' }), 'amoASenhoraTeste');
    userEvent.click(screen.getByRole('button', { name: /Entrar/i }));
    await waitFor(() => {
      expect(formSettings.handleSubmit).toHaveBeenCalledTimes(1);
      expect(formSettings.handleSubmit).toHaveBeenCalledWith({
        email: 'senhor_teste@gmail.com',
        senha: 'amoASenhoraTeste',
      },
      expect.any(Function),
      expect.any(Function),
      expect.any(Object));
    });
  });
});
