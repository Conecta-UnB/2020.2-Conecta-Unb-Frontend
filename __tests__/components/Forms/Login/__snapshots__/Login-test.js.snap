// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing loginforms must match snapshot 1`] = `
<DocumentFragment>
  <div
    class="containerLoginForm"
  >
    <div
      class="imgContainer"
    >
      <img
        alt="logoImage"
        src="[object Object]"
      />
    </div>
    <form
      action="#"
    >
      <label>
        E-mail:
        <input
          name="email"
          type="email"
          value=""
        />
      </label>
      <label>
        Senha:
        <input
          name="senha"
          type="password"
          value=""
        />
      </label>
      <button
        type="submit"
      >
        Entrar
      </button>
    </form>
  </div>
</DocumentFragment>
`;
